manta-pay/src/crypto/ecc/arkworks.rs

```diff
@@ -545,3 +539,80 @@ where
         )
     }
 }
+
+impl<C, CV> PointAdd<Compiler<C>> for GroupVar<C, CV>
+where
+    C: ProjectiveCurve,
+    CV: CurveVar<C, ConstraintField<C>>,
+{
+    type Output = Self;
+
+    fn add(&self, rhs: &Self, compiler: &mut Compiler<C>) -> Self::Output {
+        let _ = compiler;
+        let mut result = self.0.clone();
```

Comment:
Do we really need this clone? Can we not do `Self::new(&self.0 + &rhs.0)`? I believe `+` already does `+=` internally so it won't have a performance penalty.

Reply:
unfortunately `CurveVar` does not implement `Add` trait :( We can upstream this impl to arkworks in the future